apollo-federation/src/operation/merging.rs

```diff
@@ -42,12 +43,23 @@ impl<'a> FieldSelectionValue<'a> {
                 other_field.schema == self_field.schema,
                 "Cannot merge field selections from different schemas",
             );
-            ensure!(
-                other_field.field_position == self_field.field_position,
-                "Cannot merge field selection for field \"{}\" into a field selection for field \"{}\"",
-                other_field.field_position,
-                self_field.field_position,
-            );
+            if other_field.field_position != self_field.field_position {
+                return Err(SingleFederationError::InternalUnmergeableFields {
+                    message: format!(
+                        "Cannot merge field selection for field \"{}\" into a field selection for \
+                        field \"{}\". This is a known query planning bug that was silently ignored \
+                        in the old  Javascript query planner. The new Rust-native query planner \
+                        does not fix this bug at this time, but in some cases does catch when this \
```

Comment:
A minor rephrasing:
```suggestion
                        field \"{}\". This is a known query planning bug in the old, Javascript query \
                        planner that was silently ignored. The Rust-native query planner does not \
                        fix fully address this bug at this time, but in some cases does catch when this \
```

Reply:
The first sentence change sounds fine (I don't think we need the comma between "old" and "Javascript" here though since they're acting as cumulative adjectives instead of coordinate adjectives). For the second sentence, I'm hesitant to replace "does not fix" with "does not fully address", since that implies we partially address it (which we don't). I'm fine with changing it to "does not address" though.

Reply:
That works for me